containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,291 @@
 import json
 import traceback
+from copy import deepcopy
+from typing import Dict
 
 import boto3
 import botocore
 from data_platform_logging import DataPlatformLogger
-from data_platform_paths import DataProductConfig
+from data_platform_paths import (
+    BucketPath,
+    DataProductConfig,
+    JsonSchemaName,
+    get_latest_version,
+    specification_path,
+    specification_prefix,
+)
 from dataengineeringutils3.s3 import get_filepaths_from_s3_folder, read_json_from_s3
 from jsonschema import validate
 from jsonschema.exceptions import ValidationError
 
 s3_client = boto3.client("s3")
 
+glue_csv_table_input_template = {
+    "DatabaseName": "",  # add to this from data product name pulled from api path
+    "TableInput": {
+        "Name": "",  # add to this from table name pulled from api path
+        "Description": "",
+        "Owner": "owner",
+        # "Retention": 0,
+        "StorageDescriptor": {
+            "Columns": [],  # add to this from schema passed by user in api request body
+            "Location": "",  # noqa E501 add to this once data land and timestamp generated, although could be top level bucket/data/product/version/table
+            "InputFormat": "org.apache.hadoop.mapred.TextInputFormat",
+            "OutputFormat": "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat",
+            "Compressed": False,
+            "NumberOfBuckets": -1,
+            "SerdeInfo": {
+                "SerializationLibrary": "org.apache.hadoop.hive.serde2.OpenCSVSerde",
+                "Parameters": {"field.delim": ",", "escape.delim": "\\"},
+            },
+            "BucketColumns": [],
+            "SortColumns": [],
+            "Parameters": {},
+            "StoredAsSubDirectories": False,
+        },
+        "PartitionKeys": [],
+        "TableType": "EXTERNAL_TABLE",
+        "Parameters": {"classification": "csv", "skip.header.line.count": "1"},
+    },
+}
 
-def get_data_product_metadata_spec_path(version: str = "") -> str:
-    # if version is empty we'll get the latest version
-    if version == "":
-        file_paths = get_filepaths_from_s3_folder(
-            DataProductConfig.metadata_spec_prefix().uri
-        )
+
+def get_data_product_specification_path(
+    spec_type: JsonSchemaName, version: None | str = None
+) -> str:
+    """gets path for given version or latest version of metadata and schema json schema"""
+
+    # if version is empty we get the latest version
+    if version is None:
+        file_paths = get_filepaths_from_s3_folder(specification_prefix(spec_type).uri)
```

Comment:
should we use `get_latest_version` here? ([definition](https://github.com/ministryofjustice/data-platform/blob/cd6846a08234e02376780af790365ba674c04a0d/containers/daap-python-base/src/var/task/data_platform_paths.py#L193); [already imported](https://github.com/ministryofjustice/data-platform/pull/1799/files#diff-b6cacd5f702c5b90ca80f6359737c2543b425e49aecdf98c4014e09bc518f267R13))

Reply:
Is slightly different. This gets the latest version of the metadata or schema json schema file, generic to all data products and `get_latest_version` returns the latest version of a data product. Could potentially refactor `get_latest_version` so it gets the latest version of wherever it's pointed though

Reply:
Or if we don't have a single function, we could have one for each, like `get_latest_json_schema_version` vs `get_latest_data_product_version`. I think we have to be specific whenever we talk about schemas and versions or it will get very confusing 😱